### ISSUE TITLE
openmp in CMakeLists.txt needs to be updated

### ISSUE THREAD

**Issue by hubutui** [apache/incubator-mxnet#17845]
Currently, cmake check if `${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/openmp/CMakeLists.txt` exists, check [here](https://github.com/apache/incubator-mxnet/blob/master/CMakeLists.txt#L430), then build openmp from source, which is always true, and cmake will never use openmp provided in system, check [here](https://github.com/apache/incubator-mxnet/blob/master/CMakeLists.txt#L444) .  So these codes need to to reordered. Another issue, the minimum cmake version is 3.13, and according to cmake [doc](https://cmake.org/cmake/help/v3.13/module/FindOpenMP.html), `OpenMP_FOUND` is used, not `OPENMP_FOUND`, this [line](https://github.com/apache/incubator-mxnet/blob/master/CMakeLists.txt#L446) needs to be updated, too.

**Comment by leezu** [comment]
`OPENMP_FOUND` is provided by cmake for backwards compatibility and also works.

@cjolivier01 has a veto on enabling users to use system openmp. You need to manually delete `3rdparty/openmp` or modify `CMakeLists.txt` to use system openmp.

**Comment by cjolivier01** [comment]
Just remove openmp entirely.  I don’t care anymore.

**Comment by hubutui** [comment]
Yeah, I know I could remove `3rdparty/openmp`, and that's what I do for now.
